### ISSUE TITLE
[bug] Generated CMake files cause errors with `-Werror=dev --warn-uninitialized`

### ISSUE THREAD

**Issue by friendlyanon** [conan-io/conan#10288]
Conan generated CMake files cause errors when the project is configured with `-Werror=dev --warn-uninitialized`.

### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Compiler+version: VS2019
  * Conan version: 1.44.0
  * Python version: 3.10.0

### Steps to reproduce (Include if Applicable)

`find_package()` a dependency after installing it with `CMakeToolchain` and `cmake_find_package_multi` generators.

### Logs (Executed commands with output) (Include/Attach if Applicable)

Example log attached for when using the toolchain to `find_package(freetype)`. It's just a lot of uninitialized variable use.

[log.log](https://github.com/conan-io/conan/files/7833584/log.log)

**Comment by friendlyanon** [comment]
Using things like `string(APPEND|PREPEND)` and `list(APPEND|PREPEND)` will not emit this error when they operate on a variable that is not defined. So for example one could make these transformations:

```cmake
# These cause errors
set(CONAN_CXX_FLAGS "${CONAN_CXX_FLAGS} /MP4")
set(CMAKE_SHARED_LINKER_FLAGS_INIT "${CMAKE_SHARED_LINKER_FLAGS_INIT} ${CONAN_SHARED_LINKER_FLAGS}")
set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR} ... ${CMAKE_MODULE_PATH})

# These work just fine
string(APPEND CONAN_CXX_FLAGS " /MP4")
if(DEFINED CONAN_SHARED_LINKER_FLAGS)
  string(APPEND CMAKE_SHARED_LINKER_FLAGS_INIT " ${CONAN_SHARED_LINKER_FLAGS}")
endif()
list(PREPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}" ...)
```

**Comment by memsharded** [comment]
Hi @friendlyanon 

We are now not doing changes to these generators like ``cmake_find_package_multi``, and focusing our efforts on the 2.0-ready ones, as 2.0 is already in alpha stage and it is number 1 priority. Furthermore ``CMakeToolchain`` is only intended to work together with ``CMakeDeps`` not with other generators.

If you could please check if this happens too in ``CMakeDeps``, this would certainly be prioritized.

**Comment by friendlyanon** [comment]
Not near my PC right now, but the examples above are lines from a toolchain, so `CMakeDeps` having it is also likely.

**Comment by memsharded** [comment]
Thanks @friendlyanon 

yes, you are right, this belongs to CMakeToolchain, I was confused. Lets try to fix it for next release.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/10292 should address it, pointing it to next 1.45

**Comment by friendlyanon** [comment]
Note that `CMakeDeps` also causes these errors:

[log.log](https://github.com/conan-io/conan/files/7838755/log.log)

**Comment by memsharded** [comment]
Adding checks (and fixes) for CMakeDeps too.

I am struggling with a couple of vars:

```
         uninitialized variable 'dep_COMPONENT_NAMES'
E           Call Stack (most recent call first):
E             dep-config.cmake:11 (include)
E             CMakeLists.txt:7 (find_package)
E           This error is for project developers. Use -Wno-error=dev to suppress it.
E           
E           -- Conan: Target declared 'dep::dep'
E           CMake Error (dev) at dep-config.cmake:14 (foreach):
E             uninitialized variable 'dep_FIND_DEPENDENCY_NAMES'

```
they seem to be properly defined in the xxx-data.cmake files, but for some reason it keeps complaining.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/10292 was merged, it will be released in 1.45
